src/net.md

```diff
@@ -537,6 +538,71 @@ fn run() -> Result<()> {
 quick_main!(run);
 ```
 
+[ex-random-port-tcp]: #ex-random-port-tcp
+<a name="ex-random-port-tcp"></a>
+## Listen on Unused port TCP/IP
+
+[![std-badge]][std] [![cat-net-badge]][cat-net]
+
+Give the OS the responsibility to pick an unused [registered port].  These ports
+generally do not require sudo permission to bind to, and can be helpful when
+establishing connections with clients in a disposable communication chain.
+These ports begin with 1024 and have around 48,000 available.  Typically, a
+known port is used to initiate the communication and subsequent communication
+occurs on the registered port after a handshake process.
+
+In this example, the port is displayed on the console, and will listen until a
+request is made.  If you use your browser to test this program, simply enter
+the address:port into your location bar and make the request.  Because the
+program returns nothing, the browser's stop feature can be used to speed things
+up.
+
+```rust, no_run
+#[macro_use]
+extern crate error_chain;
+
+use std::net::{SocketAddrV4, Ipv4Addr, TcpListener};
+use std::io::Read;
+
+error_chain! {
+    foreign_links {
+        Io(::std::io::Error);
+    }
+}
+
+fn run() -> Result<()> {
+    let loopback = Ipv4Addr::new(127, 0, 0, 1);
+    // Assigning port 0 requests the OS to assign a free port
+    let socket = SocketAddrV4::new(loopback, 0);
+    let listen = TcpListener::bind(socket)?;
+    let port = listen.local_addr()?;
+    println!("Listening on {}, access this port to end the program", port);
+    let (mut tcp_stream, addr) = listen.accept()?;  //block  until requested
+    let mut input = String::new();
+    // read from the socket until connection closed by client.
+    let _ = tcp_stream.read_to_string(&mut input)?; //discard byte count
+    println!("Connection received! \r\n{:?} says {}", addr, input);
```

Comment:
I would drop the \r\n and just use
`println!("{:?} says {}", addr, input);`
here